ui/components/ui/survey-toast/survey-toast.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState, useContext } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchWithCache from '../../../../shared/lib/fetch-with-cache';
+import { DAY } from '../../../../shared/constants/time';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import {
+  getSelectedInternalAccount,
+  getLastViewedUserSurvey,
+  getUseExternalServices,
+  getParticipateInMetaMetrics,
+} from '../../../selectors';
+import { setLastViewedUserSurvey } from '../../../store/actions';
+import { Toast } from '../../multichain';
+
+type Survey = {
+  url: string;
+  description: string;
+  cta: string;
+  surveyId: number;
+};
+
+export function SurveyToast() {
+  const [survey, setSurvey] = useState<Survey | null>(null);
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const lastViewedUserSurvey = useSelector(getLastViewedUserSurvey);
+  const participateInMetaMetrics = useSelector(getParticipateInMetaMetrics);
+  const basicFunctionality = useSelector(getUseExternalServices);
+  const internalAccount = useSelector(getSelectedInternalAccount);
+
+  useEffect(() => {
+    if (!basicFunctionality || !internalAccount?.address) {
```

Comment:
Do we want to do this request if `participateInMetaMetrics` is off?  I don't *think* so but we should check with Hester.

Reply:
The task says that if participateInMetametrics would affect the tracking, not the survey; I'll check with Hester though.

Reply:
@jonybur we would not show the survey if `participateInMetaMetrics` is off. We would not have the MMID to check if the survey should be shown.  On the ticket I described an ambiguous "Handle case where user has not opted into MetaMetrics and does not have a MetaMetrics ID". We should ensure that not getting an MMID results in a bug. Probably best way to do that is to first check if the user has `participateInMetaMetrics` `ON`.